pdm/models/specifiers.py

```diff
@@ -203,13 +205,13 @@ def _rearrange(
         if not self.is_impossible:
             super().__init__(str(self))
 
-    def _comp_key(self):
+    def _comp_key(self) -> Tuple[Version, Version, Tuple[Version, ...]]:
         return (self._lower_bound, self._upper_bound, tuple(self._excludes))
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(self._comp_key())
 
-    def __eq__(self, other):
+    def __eq__(self, other: PySpecSet) -> bool:
```

Comment:
```suggestion
    def __eq__(self, other: Any) -> bool:
```

Reply:
I thought this one is supposed to have PySpecSet from seeing the code, but I now realized __eq__ will be used for comparison of anything. I'll fix.

```python
    def __eq__(self, other: Any) -> bool:
        if not isinstance(other, PySpecSet):
            return False
        return self._comp_key() == other._comp_key()
```

Reply:
Fixed here

[`312287c` (#354)](https://github.com/pdm-project/pdm/pull/354/commits/312287c375db739caa180500afbc976002515113#diff-8b1134a03683f14d179d12de9b62c6e35d2689782669129428ad0324c77ff0abR214)